### PR TITLE
create a DAG for crawling EU covid response timeline

### PR DESCRIPTION
**Pull request by cosletmihai**

### CODE REVIEW

File: law_fetcher/entrypoints/crawlers/eu_action_timeline/settings.py [hunk 1]
```diff
@@ -8,8 +8,12 @@
 
 # Obey robots.txt rules
 ROBOTSTXT_OBEY = True
+DOWNLOAD_DELAY = 3
+DEFAULT_REQUEST_HEADERS = {
+    'User-Agent': 'Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.36'
+}
 
-SPLASH_URL = 'http://localhost:5080'
+SPLASH_URL = 'http://srv.meaningfy.ws:5080'
```
Comment (costezki) [inline law_fetcher/entrypoints/crawlers/eu_action_timeline/settings.py:16]: is it already in the docker-elk suite? We shall make sure it is up and running.

File: law_fetcher/entrypoints/crawlers/eu_action_timeline/spiders/eu_timeline_spider.py [hunk 2]
```diff
@@ -8,26 +10,33 @@ class EUTimelineSpider(scrapy.Spider):
     name = 'eu-timeline'
     url = 'https://ec.europa.eu/info/live-work-travel-eu/coronavirus-response/timeline-eu-action_en'
```
Comment (costezki) [inline law_fetcher/entrypoints/crawlers/eu_action_timeline/spiders/eu_timeline_spider.py:11]: If these are constants, let's capitalise them, then.

File: law_fetcher/entrypoints/dags/crawl_eu_action_timeline.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    logger.info(f'Loading files from {MINIO_URL}')
+
+    minio = MinioAdapter(MINIO_URL, MINIO_ACCESS_KEY, MINIO_SECRET_KEY, MINIO_BUCKET)
+    objects = minio.list_objects(TIKA_FILE_PREFIX)
+    object_count = 0
+    for obj in objects:
+        try:
+            logger.info(f'Sending to ElasticSearch ( {ELASTICSEARCH_INDEX_NAME} ) the object {obj.object_name}')
+            elasticsearch_client.index(index=ELASTICSEARCH_INDEX_NAME,
+                                       body=loads(minio.get_object(obj.object_name).decode('utf-8')))
```
Comment (costezki) [inline law_fetcher/entrypoints/dags/crawl_eu_action_timeline.py:109]: this `loads(minio.get_object(obj.object_name).decode('utf-8'))` could be a dedicated variable to make it easier reading the code.

**Review by costezki — APPROVED** [review]
well done

File: law_fetcher/entrypoints/dags/crawl_eu_action_timeline.py (hunk 3 above)
Comment (cosletmihai) [reply law_fetcher/entrypoints/dags/crawl_eu_action_timeline.py:109]: what do you mean by dedicated variable?

File: law_fetcher/entrypoints/crawlers/eu_action_timeline/items.py [hunk 4]
```diff
@@ -11,7 +11,9 @@ class EuActionTimelineItem(scrapy.Item):
     date = scrapy.Field()
     title = scrapy.Field()
     detail_title = scrapy.Field()
-    article_content = scrapy.Field()
+    detail_link = scrapy.Field()
+    detail_content = scrapy.Field()
+    detail_metadata = scrapy.Field()
     excerpt = scrapy.Field()
```
Comment (costezki) [inline law_fetcher/entrypoints/crawlers/eu_action_timeline/items.py:17]: is this an abstract?

File: law_fetcher/entrypoints/crawlers/eu_action_timeline/items.py [hunk 5]
```diff
@@ -11,7 +11,9 @@ class EuActionTimelineItem(scrapy.Item):
     date = scrapy.Field()
     title = scrapy.Field()
     detail_title = scrapy.Field()
-    article_content = scrapy.Field()
+    detail_link = scrapy.Field()
+    detail_content = scrapy.Field()
+    detail_metadata = scrapy.Field()
```
Comment (costezki) [inline law_fetcher/entrypoints/crawlers/eu_action_timeline/items.py:16]: which metadata do we have here?

**Review by costezki — APPROVED** [review]

File: law_fetcher/entrypoints/dags/crawl_eu_action_timeline.py (hunk 3 above)
Comment (costezki) [reply law_fetcher/entrypoints/dags/crawl_eu_action_timeline.py:109]: assign the value to a variable and then use that variable as the value of the body parameter.

Comment (cosletmihai) [reply law_fetcher/entrypoints/dags/crawl_eu_action_timeline.py:109]: but that's the beauty of python, I don't see a problem with this

File: law_fetcher/entrypoints/crawlers/eu_action_timeline/items.py (hunk 4 above)
Comment (cosletmihai) [reply law_fetcher/entrypoints/crawlers/eu_action_timeline/items.py:17]: I think we're using different words to describe the same thing
what do you mean by abstract?

File: law_fetcher/entrypoints/crawlers/eu_action_timeline/items.py (hunk 5 above)
Comment (cosletmihai) [reply law_fetcher/entrypoints/crawlers/eu_action_timeline/items.py:16]: image

**Pull request merged**